Sources/Sentry/SentryUIPerformanceTracker.m

```diff
@@ -0,0 +1,258 @@
+#import "SentryUIPerformanceTracker.h"
+#import "SentryHub.h"
+#import "SentryLog.h"
+#import "SentryPerformanceTracker.h"
+#import "SentrySDK+Private.h"
+#import "SentryScope.h"
+#import "SentrySwizzle.h"
+#import "UIViewControllerHelper.h"
+#import <objc/runtime.h>
+
+#if SENTRY_HAS_UIKIT
+#    import <UIKit/UIKit.h>
+#endif
+
+static NSString *const SENTRY_UI_PERFORMANCE_TRACKER_SPAN_ID
+    = @"SENTRY_UI_PERFORMANCE_TRACKER_SPAN_ID";
+
+static NSString *const SENTRY_UI_PERFORMANCE_TRACKER_LAYOUTSUBVIEW_SPAN_ID
+    = @"SENTRY_UI_PERFORMANCE_TRACKER_LAYOUTSUBVIEW_SPAN_ID";
+
+@implementation SentryUIPerformanceTracker
+
++ (void)start
+{
+// If there`s no UIKIT don`t need to try the swizzling.
+#if SENTRY_HAS_UIKIT
+    [SentryUIPerformanceTracker swizzleViewControllerInits];
+#else
+    [SentryLog logWithMessage:@"NO UIKit -> [SentryUIPerformanceTracker "
+                              @"start] does nothing."
+                     andLevel:kSentryLevelDebug];
+#endif
+}
+
+/**
+ * Swizzle the some init methods of the view controller,
+ * so we can swizzle user view controller subclass on demand.
+ */
++ (void)swizzleViewControllerInits
+{
+#if SENTRY_HAS_UIKIT
```

Comment:
if `swizzleViewControllerInits` is only called when there's `SENTRY_HAS_UIKIT`, why would we do that?

Reply:
otherwise it does not compile